### DISCUSSION TITLE
Why is Siona RT differentiable and what algorithm does it use to resolve discontinuities

### DISCUSSION THREAD

**Discussion by JackYifan** [NVlabs/sionna#602] [Q&A]
Why is Siona RT differentiable and what algorithm does it use to resolve discontinuities

**Comment by jhoydis** [comment]
Hi @JackYifan,

Sionna RT is differentiable because it can automatically compute the derivative of a function of its output, e.g., channel impulse response or coverage map, with respect to most scene parameters, such as material properties, scattering patterns, RIS amplitude profile, position of antenna elements within an array, orientation, antenna pattern, etc.

However, discontinuities such as a change of visibility are not considered, i.e., paths cannot appear/disappear.

**Reply by JackYifan** [reply]
@jhoydis 
So sionna can't optimize antenna position? Because antenna position changes cause discontinuities such as a change of visibility. As I metioned in https://github.com/NVlabs/diff-rt/issues/7

**Reply by jhoydis** [reply]
What you cannot optimize are the position of radio devices, i.e., transmitters and receivers. However, you can optimize the relative antenna positions within an array (assuming ``scene.synthetic_array`` is True) . This assumes however that no visibility changes are introduced.

**Reply by JackYifan** [reply]
@jhoydis 
So all variables that change the path of the ray tracing are not differentiable?

**Reply by jhoydis** [reply]
Correct. This is the case for now.

**Reply by JackYifan** [reply]
@jhoydis
Thank you for your reply, could I ask by the way when sionna will be able to support refractive field calculations？

**Reply by JackYifan** [reply]
@jhoydis What's the difference between [instant-rm](https://github.com/NVlabs/instant-rm) and sionna RT. Can instant-rm optimize antenna position?
